### DISCUSSION TITLE
BUG (?): refresh and max-height

### DISCUSSION THREAD

**Discussion by nitzan-shaked** [Textualize/textual#1384] [Q&A] [answered]
(first of all: I'm happy to attach screenshots, videos, minimal code to repro, etc. -- just not sure how this community works. Apologies in advance!)

It looks like the following misbehaves:
  a `Static` with `height: auto`,
  nested in a `Vertical` with `height: auto`,
  itself nested in a `Vertical` with `height: auto; max-height: 10`

Specifically: set the context of the `Static` to something long (say 20 lines of text), which displays nicely and adds a scrollbar as expected. Then, set the content of the `Static` to "Hello" -- it works, but everything remains "too tall". A simple refresh (just resize the terminal a bit) solves this.

Given that a simple terminal resize "fixes" this I am assuming that's not intended behavior.

Specifically, changing the code of `Widget._clear_arrangement_cache` to:

```
    def _clear_arrangement_cache(self) -> None:
        """Clear arrangement cache, forcing a new arrange operation."""
        self._arrangement = None
        # THE BELOW IS NEW
        if isinstance(self._parent, Widget):
            self._parent._clear_arrangement_cache()
```

Seems to solve, but honestly I'm not sure that's the right solution. If I'm anywhere in the right direction then there should be a "force_layout" method which will do this, and set `._layout_required`, and probably do something with `Screen` (post a message? Set its layout_required? Donno.)

So, questions:

1. Am I missing the obvious? If so -- what is it?
2. And while I'm here -- are you interested in PR contribs? Eg fixing this; or -- stuff for TreeControl which I've added; or for DataTable.
3. If so -- how? Where's the guide you'd like me to follow? (tests, conventions, ci, ...)

Cheers!

**Comment by davep** [comment, marked as answer]
> I'm happy to attach screenshots, videos, minimal code to repro

Minimal code to reproduce is always a win; with accompanying steps to follow and/or screenshots/video/etc is always super helpful.

**Reply by nitzan-shaked** [reply]
Attaching `foo.py` and `foo.css`. To recreate:

1. Run `foo.py` (I chmod +x it and just `./foo.py` when both are in the same directory)
2. You get the following output:
![Screen Shot 2022-12-18 at 9 58 59](https://user-images.githubusercontent.com/1918551/208287593-8b50224c-047b-498d-80cc-cc073103bd11.png)
3. Press '1' on the keyboard. You get:
![Screen Shot 2022-12-18 at 9 59 27](https://user-images.githubusercontent.com/1918551/208287600-2a2781f0-7726-4d03-b1c9-300c4a9cf651.png)
4. Press '2'. You get: 
![Screen Shot 2022-12-18 at 9 59 47](https://user-images.githubusercontent.com/1918551/208287613-20498ad7-1ab1-444b-b8d5-9f8b6817c2e7.png)
5. Scroll, or just resize the terminal. It immediately "jumps" and becomes:
![Screen Shot 2022-12-18 at 10 00 23](https://user-images.githubusercontent.com/1918551/208287635-15b4bcac-7ecf-450c-be1e-631cc9e8a24e.png)

Is this how we report bugs? Here?
And also -- how about how to contib PRs?

**Reply by nitzan-shaked** [reply]
Ha. Forgot to attach files. Here they are.
[foo.zip](https://github.com/Textualize/textual/files/10253383/foo.zip)

**Reply by willmcgugan** [reply]
@nitzan-shaked Can you reproduce this with 0.7.0 ?

**Reply by nitzan-shaked** [reply]
@willmcgugan yes, just checked "main" on GItHub which according to `pyproject.toml` is 0.7.0.
Have you looked at my proposed fix? Regardless of whether it is correct or not, it should give a hint as to what's going on.

**Reply by nitzan-shaked** [reply]
@willmcgugan what's the next step here plz?

**Reply by willmcgugan** [reply]
There will be a fix in 0.8.0 which will be released in a day or two.

**Reply by nitzan-shaked** [reply]
Looks like it's fixed indeed. Or least: works for me :) Thank you.
